server/server.go

```diff
@@ -2073,7 +2091,7 @@ func (c *tlsMixConn) Read(b []byte) (int, error) {
 	return c.Conn.Read(b)
 }
 
-func (s *Server) createClient(conn net.Conn, ws *websocket) *client {
+func (s *Server) createClient(conn net.Conn, ws *websocket, mqtt *mqtt) *client {
```

Comment:
Feels like `createClient` should stay the same and have specialized version for ws and mqtt vs passing them in.

Reply:
There would be a LOT of code to copy/paste and have to fix issues in 3 different places (we have been fixing issues in that code very recently).

Reply:
Done.